core/capabilities/ccip/ocrimpls/contract_transmitter.go

```diff
@@ -23,54 +23,63 @@ type ToCalldataFunc func(
 	report ocr3types.ReportWithInfo[[]byte],
 	rs, ss [][32]byte,
 	vs [32]byte,
-) (any, error)
+) (string, string, any, error)
+
+func NewToCommitCalldata(defaultMethod, priceOnlyMethod string) ToCalldataFunc {
+	return func(
+		rawReportCtx [2][32]byte,
+		report ocr3types.ReportWithInfo[[]byte],
+		rs, ss [][32]byte,
+		vs [32]byte,
+	) (string, string, any, error) {
+		// Note that the name of the struct field is very important, since the encoder used
+		// by the chainwriter uses mapstructure, which will use the struct field name to map
+		// to the argument name in the function call.
+		// If, for whatever reason, we want to change the field name, make sure to add a `mapstructure:"<arg_name>"` tag
+		// for that field.
+		var info ccipocr3.CommitReportInfo
+		if len(report.Info) != 0 {
+			var err error
+			info, err = ccipocr3.DecodeCommitReportInfo(report.Info)
+			if err != nil {
+				return "", "", nil, err
+			}
+		}
 
-func ToCommitCalldata(
-	rawReportCtx [2][32]byte,
-	report ocr3types.ReportWithInfo[[]byte],
-	rs, ss [][32]byte,
-	vs [32]byte,
-) (any, error) {
-	// Note that the name of the struct field is very important, since the encoder used
-	// by the chainwriter uses mapstructure, which will use the struct field name to map
-	// to the argument name in the function call.
-	// If, for whatever reason, we want to change the field name, make sure to add a `mapstructure:"<arg_name>"` tag
-	// for that field.
-	var info ccipocr3.CommitReportInfo
-	if len(report.Info) != 0 {
-		var err error
-		info, err = ccipocr3.DecodeCommitReportInfo(report.Info)
-		if err != nil {
-			return nil, err
+		method := defaultMethod
+		if len(priceOnlyMethod) > 0 && len(info.MerkleRoots) == 0 && len(info.TokenPrices) > 0 {
```

Comment:
Doing `len(string)` seems awkward/non-idiomatic, I feel like `priceOnlyMethod != ""` reads better.

Also, if we always require a merkle root then shouldn't `len(info.MerkleRoots) == 1`? cc @archseer

Reply:
Just making a note that you can't tweak this array here since it'll fail the sig check onchain, this has to come from the plugin.

Reply:
Presumably it's invalid to have a report where `len(info.MerkleRoots) == 0 && len(info.TokenPrices) == 0`.

|  onchain method  | HasRoot(s) | HasPrices |
|---|----------------|---------------|
|  execute | true | true |
| execute  | true  | false |
|  price_only_execute | false | true |
| N/A (should we return error?)  | false | false |

I think the report codec should detect that instead of the transmitter.